middleware/grpc_instrumentation.go

```diff
@@ -173,18 +171,115 @@ func (s *instrumentedClientStream) CloseSend() error {
 	return err
 }
 
-// errorCode converts an error into an error code string.
-func errorCode(err error) string {
-	if err == nil {
-		return "2xx"
+type InstrumentationLabelOption func(*InstrumentationLabel)
+
+var (
+	AcceptGRPCStatusesOption InstrumentationLabelOption = func(instrumentationLabel *InstrumentationLabel) {
+		instrumentationLabel.AcceptGRPCStatuses(true)
 	}
 
-	if errResp, ok := httpgrpc.HTTPResponseFromError(err); ok {
-		statusFamily := int(errResp.Code / 100)
-		return strconv.Itoa(statusFamily) + "xx"
-	} else if grpcutil.IsCanceled(err) {
+	DoNotAcceptGRPCStatusesOption InstrumentationLabelOption = func(instrumentationLabel *InstrumentationLabel) {
+		instrumentationLabel.AcceptGRPCStatuses(false)
+	}
+)
+
+func ApplyInstrumentationLabelOptions(maskHTTPStatuses bool, options ...InstrumentationLabelOption) InstrumentationLabel {
+	instrumentationLabel := InstrumentationLabel{maskHTTPStatuses: maskHTTPStatuses}
+	for _, opt := range options {
+		opt(&instrumentationLabel)
+	}
+	return instrumentationLabel
+}
+
+type InstrumentationLabel struct {
+	acceptGRPCStatuses bool
+	maskHTTPStatuses   bool
+}
+
+func (i *InstrumentationLabel) AcceptGRPCStatuses(acceptance bool) {
+	i.acceptGRPCStatuses = acceptance
+}
+
+// GetInstrumentationLabel converts an error into an error code string by applying the configurations
+// contained in this InstrumentationLabel object.
+func (i *InstrumentationLabel) GetInstrumentationLabel(err error) string {
+	statusCode := i.errorToStatusCode(err)
+	return i.statusCodeToString(statusCode)
+}
+
+// statusCodeToString converts a given status code in its string representation.
+//
+//   - If the given status code is codes.OK, statusCodeToString returns "success" or "2xx",
+//     depending on whether maskHTTPStatuses is set to false or true respectively.
+//   - If the given status code is codes.Canceled, statusCodeToString returns "cancel"
+//     independently of maskHTTPStatuses.
+//   - If the given status code ia a valid HTTP status code, statusCodeToString returns the
+//     string representation of the HTTP status code or the first digit of the HTTP status
+//     code followed by "xx", depending on whether maskHTTPStatuses is set to false or true
+//     respectively.
+//   - If the given status code is a gRPC status code different from codes.Unknown, statusCodeToString
+//     returns the string representation of the status code, independently of maskHTTPStatuses.
+//   - If the given status code is codes.Unknown, statusCodeToString returns "error".
+func (i *InstrumentationLabel) statusCodeToString(statusCode codes.Code) string {
+	if statusCode == codes.OK {
+		if i.maskHTTPStatuses {
+			return "2xx"
+		}
+		return "success"
+	}
+
+	if statusCode == codes.Canceled {
 		return "cancel"
-	} else {
+	}
+
+	if statusCode == codes.Unknown {
 		return "error"
 	}
```

Comment:
I think we should be consistent, and report `OK`, `Canceled` and `Unknown`, when reporting of gRPC statuses is enabled. (or perhaps lowercase version)

When it's disabled, we can go back to `success`, `cancel` or `error`.

WDYT?

Reply:
I have applied your recommendation. I haven't opted for lowercase version of gRPC status codes, because for the majority of them the original form is more readable: e.g., `FailedPrecondition` vs. `failedprecondition`, `ResourceExhausted` vs. `resourceexhausted`, etc.

Reply:
Looks good, thanks.